### PR TITLE
Upgrade CoffeeScriptLinter to broccoli-persistent-filter

### PR DESCRIPTION
**Pull request by johnnyshields**
This is the latest API from Ember team (see https://github.com/ember-cli/ember-cli/issues/5030) and the change below cuts my build time in half. I've tested this on a large CoffeeScript ember project.

### CODE REVIEW

**Comment by johnnyshields** [comment]
@kimroen please look at this, makes a significant improvement in build time.

**Comment by jonnii** [comment]
:+1:

**Comment by stefanpenner** [comment]
just made quick review, this LGTM

**Comment by stefanpenner** [comment]
ah, this should likely enable `persist: true` so fast warm boots are possible

File: lib/coffee-linter.js [hunk 1]
```diff
@@ -12,6 +14,7 @@ function CoffeeScriptLinter (inputTree, options) {
   if (!(this instanceof CoffeeScriptLinter)) return new CoffeeScriptLinter(inputTree, options);
   Filter.call(this, inputTree, options);
   options = options || {};
+  this.options = options;
```
Comment (kimroen) [inline lib/coffee-linter.js]: Could you set the `options.persist = true`, and then move the `Filter.call` line to after this?

**Comment by kimroen** [comment]
All right, tried all this out and it's looking really good :) Would you mind making the change above? After that I'll try out again to confirm, and then merge and release. Hopefully I can get that done tomorrow :)

**Comment by kimroen** [comment]
Thanks, and thanks for your help, stef :smile:

**Comment by johnnyshields** [comment]
@kimroen done.

**Comment by johnnyshields** [comment]
@kimroen the change as suggested broke our build. Can we merge this PR as-is and then implement this persist option separately, as I don't know a lot about this `.persist` option and I'm really not the right person to implement this.

**Comment by johnnyshields** [comment]
FYI here's the log which occurred on ref ab59e5434bc534e5dad428c7a4fa7f07cb1a041e

```
BuildingCannot find module '/workspace/node_modules/ember-cli-coffeescript/lib//package.json'
Error: Cannot find module '/workspace/node_modules/ember-cli-coffeescript/lib//package.json'
  at Function.Module._resolveFilename (module.js:336:15)
  at Function.Module._load (module.js:278:25)
  at Module.require (module.js:365:17)
  at require (module.js:384:17)
  at pkg (/workspace/node_modules/ember-cli-coffeescript/node_modules/broccoli-persistent-filter/node_modules/hash-for-dep/lib/pkg.js:19:20)
  at statPathsFor (/workspace/node_modules/ember-cli-coffeescript/node_modules/broccoli-persistent-filter/node_modules/hash-for-
```

**Comment by stefanpenner** [comment]
looks like hash-for-dep is unable to resolve coffeescripts `package.json`

**Comment by johnnyshields** [comment]
@stefanpenner can we make that a separate issue please and merge this as-is.

**Comment by stefanpenner** [comment]
> @stefanpenner can we make that a separate issue please and merge this as-is.

im indifferent (and also not a maintainer of the project, so can't really merge either way :P)

opened an issue, will investigate the failure you describe soon (maybe tomorrow) ->https://github.com/stefanpenner/hash-for-dep/issues/21

**Comment by johnnyshields** [comment]
@kimroen please merge?

**Comment by thomassnielsen** [comment]
@johnnyshields could you please stop nagging? This project (as most open source projects) is maintained by volunteers, and as such may have limited time and have no obligation to look at issues and pull requests this often. It will be looked at in time. If you need this change, you can run your own fork of the project until it is considered and possibly merged.

**Comment by kimroen** [comment]
Took a brief look at the error, seems likely that this part is causing the problem:

```js
CoffeeScriptLinter.prototype.baseDir = function() {
  return __dirname;
};
```

This file is not in the root of the project, so the `package.json` file wouldn't be there. Maybe this:

```js
CoffeeScriptLinter.prototype.baseDir = function() {
  return path.resolve(__dirname, '..');
};
```

I'll merge and try that later today.

**Comment by stefanpenner** [comment]
@kimroen good catch!

**Comment by johnnyshields** [comment]
Tried @kimroen's fix, but looks like it's getting a different error:

```
version: 1.13.8
Cannot find module 'timers-ext/' from '\workspace\node_modules\ember-cli\node_modules\inquirer\node_modules\cli-color\node_modules\memoizee\'

Error: Cannot find module 'timers-ext/' from '\workspace\node_modules\ember-cli\node_modules\inquirer\node_modules\cli-color\node_modules\memoizee\'
  at Function.module.exports (\workspace\node_modules\ember-cli-coffeescript\node_modules\broccoli-persistent-filter\node_modules\hash-for-dep\node_modules\resolve\lib\sync.js:33:11)
  at resolvePkg (\workspace\node_modules\ember-cli-coffeescript\node_modules\broccoli-persistent-filter\node_modules\hash-for-dep\lib\resolve-pkg.js:19:18)
  at pkg (\workspace\node_modules\ember-cli-coffeescript\node_modules\broccoli-persistent-filter\node_modules\hash-for-dep\lib\pkg.js:17:20)
  at again (\workspace\node_modules\ember-cli-coffeescript\node_modules\broccoli-persistent-filter\node_modules\hash-for-dep\lib\deps-for.js:18:22)
  at \workspace\node_modules\ember-cli-coffeescript\node_modules\broccoli-persistent-filter\node_modules\hash-for-dep\lib\deps-for.js:27:7
  at Array.forEach (native)
```

**Comment by kimroen** [comment]
Merging this now, then I'll take a look at the problem with `persist`, and then release with or without it depending.

**Comment by kimroen** [comment]
Published without the persist option as `v.1.13.2`. Thanks again for your help, @johnnyshields!

**Pull request merged**